invictus_dashboard/src/app/shared/security/services/security.service.spec.ts

```diff
@@ -1,16 +1,74 @@
+import { HttpClient } from '@angular/common/http';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
 import { TestBed } from '@angular/core/testing';
-
+import { RouterTestingModule } from '@angular/router/testing';
+import { SharedModule } from '../../shared.module';
+import { AppUser } from '../app-user';
+import { AppUserAuth } from '../app-user-auth';
+import { UsersData } from '../users-data';
 import { SecurityService } from './security.service';
 
 describe('SecurityService', () => {
   let service: SecurityService;
+  let httpClient: HttpClient;
+  let httpTestingController: HttpTestingController;
+  let testUserList: AppUser[];
 
   beforeEach(() => {
-    TestBed.configureTestingModule({});
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule, SharedModule, RouterTestingModule],
+    });
+
+    httpClient = TestBed.inject(HttpClient);
+    httpTestingController = TestBed.inject(HttpTestingController);
     service = TestBed.inject(SecurityService);
+    testUserList = UsersData;
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  it('should get inital list of users with a GET HTTP req', () => {
+    service.getUsers().subscribe({
```

Comment:
don't use the next and error version of subscribe. It's deprecated. Subscribe should just be the function for success and you use `catchError` in a pipe if you want to handle an error.

Make sure you unsubscribe with something like a `take(1)`

Reply:
There is a `take(1)` in the service method being called, so it should unsubscribe after one call. I didn't realize that next was depreciated! I'll comb through and swap those over.